src/internal/operations/restore.go

```diff
@@ -355,41 +356,9 @@ func formatDetailsForRestoration(
 		return nil, err
 	}
 
-	var (
-		fdsPaths  = fds.Paths()
-		paths     = make([]path.RestorePaths, len(fdsPaths))
-		shortRefs = make([]string, len(fdsPaths))
-		el        = errs.Local()
-	)
-
-	for i := range fdsPaths {
-		if el.Failure() != nil {
-			break
-		}
-
-		p, err := path.FromDataLayerPath(fdsPaths[i], true)
-		if err != nil {
-			el.AddRecoverable(clues.
-				Wrap(err, "parsing details path after reduction").
-				WithMap(clues.In(ctx)).
-				With("path", fdsPaths[i]))
-
-			continue
-		}
-
-		dir, err := p.Dir()
-		if err != nil {
-			el.AddRecoverable(clues.
-				Wrap(err, "getting restore directory after reduction").
-				WithClues(ctx).
-				With("path", fdsPaths[i]))
-
-			continue
-		}
-
-		paths[i].StoragePath = p
-		paths[i].RestorePath = dir
-		shortRefs[i] = p.ShortRef()
+	paths, err := pathtransformer.GetPaths(ctx, backupVersion, fds.Items(), errs)
```

Comment:
suggestion (can't comment on line 347): how about moving this whole func into `pathtransformer`, and exporting it, instead of exporting GetPaths?

Reply:
hmmm I thought about that, but I kind of like the separation we have here. Keeping the reduce operation separate from the path transform stuff gives us a nice interface for the path transform stuff that we can test on. That code now has one job only: take a set of known input items and generate a set of paths based on well-defined transforms. If we pull the reduce operation into that then the path transform abstraction (and the interface-level testing) for it get more complex because we need to ensure both that reduce works as expected and that the path transformations work as expected

If we had mocks for selectors then maybe I'd be more willing to combine them, but right now if we combine things the tests would all have to setup the proper type of selector, make sure the selector reduce produced expected results, and then setup/run the path transform stuff. That's quite a bit of work. Sure, we could start splitting the tests, but then we're looking more at white box testing (testing internal functions) rather than black box testing (testing interface behavior). White box testing can lead to much more churn and a higher cost to modifying how things work internally which I'd kind of like to avoid

Reply:
All in all I'm fine with this decision.  Just a nitpick on the side:

> If we had mocks for selectors then maybe I'd be more willing to combine them, but right now...

You don't need mocks on selectors for that.  You just need an interface to mock. One such as [this interface](https://github.com/alcionai/corso/blob/main/src/pkg/selectors/selectors.go#L72).  The func accepts the interface, and callers can either provide a selector (prod case) or a mock (test case).  That's only a sliver of the effort being suggested.